### PR TITLE
feat(model): add method wrapper to define the api response wrapper / overrides

### PR DESCRIPTION
**Pull request by JRogaishio**
## Checklist

- [x] Tests
- [x] Docs
- [x] Type definitions 

**Changes:**
- Added wrap, _unwrap, wrappedBy, and nowrap methods to Model.js to mimic Laravel's resource data wrapping. This makes for `$first`, `$find`, `$get`, and `$all` obsolete but I kept them for backward compatibility. The the overloadable `wrap()` allows you to automatically unwrap using the standard `first`, `find`, `get`, `all` methods but also change what the wrapper key is in case you don't want to use the standard 'data' property
- Lock the defu package to 6.1.2. Version 6.1.3+ has a regression that prevents exported objects from merging with plain objects.
- Added unit tests for the new wrap() method
- Added new `wrap()` type
- Added new `wrappedBy(wrap)` and `nowrap()` builder methods to change / disable wrapping
- Updated documentation for new `wrap()` type
- Updated documentation for new `wrappedBy()` and `nowrap()` methods

### CODE REVIEW

File: package-lock.json
Comment (HashGateApp) [inline package-lock.json:1]: This file should be excluded from the commit

File: yarn.lock
Comment (HashGateApp) [inline yarn.lock:1]: This file should be excluded from the commit

Comment (JRogaishio) [reply yarn.lock:1]: Resolved. yarn.lock has been removed and also added to the `.gitignore` so it doesn't happen again.

**Comment by JRogaishio** [comment]
I updated this PR to also include `.nowrap()` and `wrappedBy(v)` methods since sometimes (Like for paginated responses) you want to disable wrapping at runtime for a single response.

**Comment by robsontenorio** [comment]
cc @JoaoPedroAS51

**Comment by HashGateApp** [comment]
@robsontenorio Who should be the reviewer of this PR?

**Comment by JoaoPedroAS51** [comment]
Hi guys, sorry for the delay. I've been very busy lately. I'll review the PR.

@JRogaishio thank you for the PR! :)

**Comment by robsontenorio** [comment]
@JRogaishio  On sidebar you need to allow edit/commits by maintainers.

So @JoaoPedroAS51  can go ahead with this.

image

**Comment by JRogaishio** [comment]
@robsontenorio I don't think I can give maintainer access to my fork / pr since it's on an organization and not **user owned**. The option just doesn't exist on the PR (see below screenshot). I did however resolve the merge conflicts and brought yarn.lock back since it appears you're using it.
image

Docs I'm trying to follow:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

> **On user-owned forks,** if you want to allow anyone with push access to the upstream repository to make changes to your pull request, select Allow edits from maintainers.

**Comment by JoaoPedroAS51** [comment]
Hi @JRogaishio! Thank you for fixing the conflict! Yeah, we only save the `yarn.lock` in the repo, which is the package manager we use for this project. Can you just revert the revert the `.gitignore`? Then I think we are good to go. Many thanks! :)

**Comment by JRogaishio** [comment]
@JoaoPedroAS51 You're welcome. Also the .gitignore has been reverted back to the robsontenorio/dev version so it should be all set now.

**Review by JoaoPedroAS51 — APPROVED** [review]

**Comment by JoaoPedroAS51** [comment]
Hi guys, sorry for not releasing it yet. It seems that the release action failed, I'll try to take a look over the weekend.

**Pull request merged**